src/components/Visualization/Visualization.js

```diff
@@ -133,26 +123,8 @@ export const Visualization = ({
         }
     }, [data, visualization])
 
-    if (error) {
-        let output
-        if (error.details?.errorCode) {
-            switch (error.details.errorCode) {
-                case 'E7205':
-                    output = noPeriodError()
-                    break
-                case 'E7132':
-                    output = indicatorError()
-                    break
-                case 'E7121':
-                    output = dataAccessError()
-                    break
-                default:
-                    output = genericServerError()
-            }
-        } else {
-            output = genericServerError()
-        }
-        dispatch(acSetLoadError(output))
+    if (error && onError) {
+        onError(error)
```

Comment:
Could be error && onError && onError(error) on a single line, but keep it if you prefer the readability.